private/tst/avb_streamhandler/src/IasTestSystemdWatchdog.cpp

```diff
@@ -0,0 +1,233 @@
+#include <stdlib.h>
+
+#include "gtest/gtest.h"
+#include "dlt/dlt.h"
+
+#define private public
+#define protected public
+#include "avb_watchdog/IasSystemdWatchdogManager.hpp"
+#include "avb_watchdog/IasWatchdogTimerRegistration.hpp"
+#include "avb_watchdog/IasWatchdogInterface.hpp"
+#include "avb_watchdog/IasWatchdogResult.hpp"
+#undef protected
+#undef private
+
+namespace IasWatchdog {
+
+class IasTestSystemdWatchdog : public ::testing::Test
+{
+  protected:
+    IasTestSystemdWatchdog()
+    : mDltContext()
+    , mWatchdogTimerRegistration(nullptr)
+    , mWdManager(nullptr)
+    {}
+
+  virtual ~IasTestSystemdWatchdog()
+  {}
+
+  virtual void SetUp()
+  {
+    mWdManager = new (std::nothrow) IasSystemdWatchdogManager(mDltContext);
+    ASSERT_TRUE(mWdManager != nullptr);
+
+    mWatchdogTimerRegistration = std::make_shared<IasWatchdog::IasWatchdogTimerRegistration>(mDltContext);
+    ASSERT_TRUE(mWatchdogTimerRegistration != nullptr);
+
+  }
+
+  virtual void TearDown()
+  {
+    delete mWdManager;
+    mWatchdogTimerRegistration = nullptr;
+  }
+
+  DltContext mDltContext;
+  std::shared_ptr<IasWatchdogTimerRegistration> mWatchdogTimerRegistration;
+  IasSystemdWatchdogManager *mWdManager;
+};
+
+TEST_F(IasTestSystemdWatchdog, init)
+{
+  ASSERT_EQ(setenv("WATCHDOG_USEC", "30000000", 1), 0);
+  IasWatchdogResult result = mWdManager->init(mWatchdogTimerRegistration);
+  ASSERT_EQ(result, IasMediaTransportAvb::IasResult::cOk);
```

Comment:
Maybe use `EXPECT_EQ` here? This way we can keep on testing other scenarios. Note that the `ASSERT_EQ` on the `setenv` call is ok, as that call is not what we are testing, and if it fails, it compromises the result of the test.